extensions/ql-vscode/src/compare/compare-interface.ts

```diff
@@ -0,0 +1,270 @@
+import { DisposableObject } from '@github/codeql-vscode-utils';
+import {
+  WebviewPanel,
+  ExtensionContext,
+  window as Window,
+  ViewColumn,
+  Uri,
+} from 'vscode';
+import * as path from 'path';
+
+import { tmpDir } from '../run-queries';
+import { CompletedQuery } from '../query-results';
+import {
+  FromCompareViewMessage,
+  ToCompareViewMessage,
+  QueryCompareResult,
+} from '../interface-types';
+import { Logger } from '../logging';
+import { CodeQLCliServer } from '../cli';
+import { DatabaseManager } from '../databases';
+import { getHtmlForWebview, jumpToLocation } from '../interface-utils';
+import { adaptSchema, adaptBqrs, RawResultSet } from '../adapt';
+import { BQRSInfo } from '../bqrs-cli-types';
+import resultsDiff from './resultsDiff';
+
+interface ComparePair {
+  from: CompletedQuery;
+  to: CompletedQuery;
+}
+
+export class CompareInterfaceManager extends DisposableObject {
+  private comparePair: ComparePair | undefined;
+  private panel: WebviewPanel | undefined;
+  private panelLoaded = false;
+  private panelLoadedCallBacks: (() => void)[] = [];
+
+  constructor(
+    private ctx: ExtensionContext,
+    private databaseManager: DatabaseManager,
+    private cliServer: CodeQLCliServer,
+    private logger: Logger,
+    private showQueryResultsCallback: (
+      item: CompletedQuery
+    ) => Promise<void>
+  ) {
+    super();
+  }
+
+  async showResults(
+    from: CompletedQuery,
+    to: CompletedQuery,
+    selectedResultSetName?: string
+  ) {
+    this.comparePair = { from, to };
+    this.getPanel().reveal(undefined, true);
+
+    await this.waitForPanelLoaded();
+    const [
+      commonResultSetNames,
+      currentResultSetName,
+      fromResultSet,
+      toResultSet,
+    ] = await this.findCommonResultSetNames(
+      from,
+      to,
+      selectedResultSetName
+    );
+    if (currentResultSetName) {
+      await this.postMessage({
+        t: 'setComparisons',
+        stats: {
+          fromQuery: {
+            // since we split the description into several rows
```

Comment:
It's not clear to me that the user-defined label case works well here. When I set the label of the query-being-compared to to `[%t] custom`, what I see is just the query file name and not the string 'custom' appear anywhere in the view.

Reply:
Hmmm...let me see.

Reply:
Ugh...silly mistake. Fixed now.